node/src/Pos/DB/Block.hs

```diff
@@ -113,6 +114,13 @@ blkGetHeader
     => HeaderHash -> m (Maybe (BlockHeader ssc))
 blkGetHeader = dbGetBi BlockIndexDB . blockIndexKey
 
+blkGetHeaderOrThrow
+    :: (HasCoreConstants, SscHelpersClass ssc, MonadDBRead m)
+    => HeaderHash -> m (BlockHeader ssc)
+blkGetHeaderOrThrow = maybeThrow (DBMalformed msg) <=< blkGetHeader
+  where
+    msg = "blkGetHeaderOrThrow: expected the header to be present in the DB, but it isn't"
```

Comment:
It would be good to also put header's hash into error message (you won't be able to use cool `<=<`, but that's because life is hard).

Reply:
Now that you told me about `getTipHeader`, I removed `blkGetHeaderOrThrow` (it's not needed in this PR anymore).